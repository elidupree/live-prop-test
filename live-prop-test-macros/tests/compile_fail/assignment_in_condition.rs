use live_prop_test::live_prop_test;

#[live_prop_test(postcondition = "result = ()")]
fn function1() {}

#[live_prop_test(precondition = "a = b")]
fn function2(a: i32, b: i32) -> i32 {b}

#[live_prop_test(precondition = "a == b")]
#[live_prop_test(postcondition = "(a != result) && (b = result)")]
fn function3(a: i32, b: i32) -> i32 {b}

fn main() {}
