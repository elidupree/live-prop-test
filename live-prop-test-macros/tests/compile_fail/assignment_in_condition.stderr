error: assignment isn't allowed in live-prop-test conditions; did you mean `==` instead of `=`?
 --> $DIR/assignment_in_condition.rs:3:34
  |
3 | #[live_prop_test(postcondition = "result = ()")]
  |                                  ^^^^^^^^^^^^^

error: assignment isn't allowed in live-prop-test conditions; did you mean `==` instead of `=`?
 --> $DIR/assignment_in_condition.rs:6:33
  |
6 | #[live_prop_test(precondition = "a = b")]
  |                                 ^^^^^^^

error: assignment isn't allowed in live-prop-test conditions; did you mean `==` instead of `=`?
  --> $DIR/assignment_in_condition.rs:10:34
   |
10 | #[live_prop_test(postcondition = "(a != result) && (b = result)")]
   |                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
